### PR TITLE
intrinsify `abs` on floats

### PR DESCRIPTION
**Pull request by soc**

### CODE REVIEW

File: dora/src/masm/x64.rs [hunk 1]
```diff
@@ -1148,6 +1148,41 @@ impl MacroAssembler {
         }
     }
 
+    pub fn float_abs(&mut self, mode: MachineMode, dest: FReg, src: FReg) {
+        let (fst, snd) = if mode == MachineMode::Float32 {
+            (0, 0x7fffffff)
+        } else {
+            (0x7fffffff, -1)
```
Comment (soc) [inline dora/src/masm/x64.rs:1155]: I think the numbers aren't correct yet, so I didn't quite get how they were derived in `float_neg` either.

File: dora/src/masm/x64.rs [hunk 2]
```diff
@@ -1148,6 +1148,41 @@ impl MacroAssembler {
...
+        }
+
+        if dest != src {
+            self.copy_freg(mode, dest, src);
+        }
+    }
+
     pub fn float_neg(&mut self, mode: MachineMode, dest: FReg, src: FReg) {
         let (fst, snd) = if mode == MachineMode::Float32 {
             (1i32 << 31, 0)
```
Comment (soc) [inline dora/src/masm/x64.rs:1188]: @dinfuehr Can you comment on how you arrived at these bits? I tried to implement it similarly for `float_abs`, but all combinations I try either produce the wrong result or crash the runtime altogether.

**Comment by soc** [comment]
@dinfuehr could you have a look?

File: dora/src/cannon/codegen.rs [hunk 3]
```diff
@@ -679,6 +679,21 @@ impl<'a> CannonCodeGen<'a> {
         self.emit_store_register(REG_RESULT.into(), dest);
     }
 
+    fn emit_abs_float(&mut self, dest: Register, src: Register) {
```
[the remark this answers is not in the document]
Comment (soc) [reply dora/src/cannon/codegen.rs:682]: shall I rename the other methods `emit_neg_float` etc. too?

**Comment by soc** [comment]
@dinfuehr could you have a look?

**Comment by dinfuehr** [comment]
Thanks!

**Comment by soc** [comment]
👍

**Pull request merged**